### ISSUE TITLE
Add guidance about not using excessively long hint text

### ISSUE THREAD

**Issue by timpaul** [alphagov/govuk-design-system#1671]
## Related documentation

https://design-system.service.gov.uk/components/text-input/#hint-text

## Suggestion

Add guidance explaining that you shouldn't use the hint text component for long paragraphs and lists, because it will cause it to be read out whenever a screen reader user interacts with the associated form element, which could be very frustrating if it's verbose.

Provide an example of how to lay out questions that require lots of explanation (like the example image below), without using the hint text component (which adds the aria-describedby attributes that cause the behaviour).

## Evidence (where applicable)

A user has observed multiple services where the hint text contains multiple paragraphs and lists (see example image):

![image](https://user-images.githubusercontent.com/1590604/118519327-ed26fa80-b730-11eb-8b7a-6ecf16a1e81f.png)

**Comment by timpaul** [comment]
This issue raised on behalf of @craigabbot

**Comment by abbott567** [comment]
@EoinShaughnessy This could definitely be solved by using an alternative pattern or restructuring the design. But I guess coming up with what is a good alternative isn't the point for this issue.

This isn't an isolated incident. We've seen a few instances of hint text being abused so it would suggest it needs to be clearer in the guidance how to use it and the impact for screen reader users called out, because I guess you don't know what you don't know.

**Comment by EoinShaughnessy** [comment]
Thanks, @abbott567 - that helps with revising the content!

**Comment by htmlandbacon** [comment]
One additional thing that isn't shown in this example, was the use of links in hint text, this got flagged on xgov slack on May the 10th but the threat has been eaten.

**Comment by EoinShaughnessy** [comment]
@htmlandbacon Thanks! We can add content about links. (To say that screen readers read out the link text as a part of the field description - but it's not clear the text is a link.)

**Comment by EoinShaughnessy** [comment]
It feels like there are 2 issues here, rather than one:
- explaining why to avoid long hint text
- providing an example of how to structure complex questions without using hint text

For the first issue, we can quickly update the hint text guidance.
For the second issue, we probably need to update the question pages pattern, and then link it to the hint text guidance.

Maybe the best way forward is to update the hint text guidance for now, and meanwhile I'll create an issue to start off the pattern update.
